pkg/devfile/parser/data/1.0.0/components.go

```diff
@@ -6,35 +6,216 @@ import (
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 )
 
-// GetComponents returns the slice of DevfileComponent objects parsed from the Devfile
+func (d *Devfile100) GetMetadata() common.DevfileMetadata {
+	// No GenerateName field in V2
```

Comment:
This part is confusing. There is no GenerateName in V2. But there *is* in V1. So we would need to still return GenerateName here as this is the `1.0.0` parser (hence the `1.0.0` dir). So we'd most likely need to include it.

Reply:
To add `generateName` field here, i need to add it common types. currently we have updated our code not to use any v1 specific fields,  `generateName` is currently used by `odo catalog list components` that has a parser of its own, so adding it here would of no use.

Reply:
the `odo catalog list components` uses `generateName` to figure out the component type i.e. nodejs. how would that change for v2 then?

Reply:
thats what i said, `odo catalog list components` has parser of its own, so no affect to add it here

Reply:
Honestly I still think we should add GenerateName here... This is fairly confusing as this is the v1.0.0 parser folder but we are implementing v2 features.